### PR TITLE
Accessory View and Accessory Type support for iOS8

### PR DESCRIPTION
**Pull request by AnthonyMDev**

### CODE REVIEW

File: AutoLayoutCells/TableViewCells/ALTableViewCellFactory.m [hunk 1]
```diff
@@ -161,6 +165,35 @@ - (CGFloat)calculateHeightForConfiguredSizingCell:(UITableViewCell *)sizingCell
...
+
+- (BOOL)isOS8OrLater
+{
+  return IS_OS_8_OR_LATER;
+}
+
+- (CGFloat)adjustWidth:(CGFloat)width forAccessoryViewOnSizingCell:(UITableViewCell *)sizingCell
+{
+  if (sizingCell.accessoryView) {
+    width -= kAccessoryViewTrailingMarginWidth;
```
Comment (JRG-Developer) [inline AutoLayoutCells/TableViewCells/ALTableViewCellFactory.m]: I'm not a fan of the magic numbers here... can we get the width from the `accessoryView` instead?

File: AutoLayoutCells/TableViewCells/ALTableViewCellFactory.m [hunk 2]
```diff
@@ -161,6 +165,35 @@ - (CGFloat)calculateHeightForConfiguredSizingCell:(UITableViewCell *)sizingCell
...
+    
+  } else if (sizingCell.accessoryType != UITableViewCellAccessoryNone) {
+    width -= kAccessoryTypeTrailingMarginWidth;
+
```
Comment (JRG-Developer) [inline AutoLayoutCells/TableViewCells/ALTableViewCellFactory.m]: I'm not a fan of the magic numbers here... can we get the width from the `accessoryView` instead?

**Comment by AnthonyMDev** [comment]
I'm not a fan of this either. I recognize that these numbers might change in later versions and this proposes a problem.

The issue here is that the `width` isn't the actual width of the `accessoryView` it seems that the width of the actual view is being accounted for. What is not being accounted for is the margin between the `accessoryView.trailing` and `UITableViewCell.trailing`.

I don't know of any way to actually access a variable for that space or to compute it. From my research and experimentation, `10.0f` for `accessoryType` and `15.0f` for `accessoryView` is the "Magic Number".

**Comment by AnthonyMDev** [comment]
@JRG-Developer If you have any suggestions or want to discuss how to do this differently, give me a call!

**Comment by AnthonyMDev** [comment]
I've done some more experimentation, and changing other constraints on the cell seems to alter this number. I'm trying to see if I can reverse engineer the formula. I'll close this and make another pull request if I can get it.

**Comment by JRG-Developer** [comment]
Yep, this can indeed be frustrating. :/

Fortunately, iOS 8 provides support from Apple for calculating dynamic table view cell size! Perhaps, this issue will just be put off until iOS 7 support can be dropped?

:]

**Pull request closed without merging**